compose/service.py

```diff
@@ -623,6 +623,10 @@ def start_container(self, container, use_network_aliases=True):
         try:
             container.start()
         except APIError as ex:
+            if "driver failed programming external connectivity" in ex.explanation:
+                log.warn(
+                    "Port is already in use, check the docker-compose file to see if the same" +
```

Comment:
I think that's a good think to check for this specific scenario, but the message here is incorrect : port collision may be the result of any other container running on same host, and not under control of this specific compose file.

Reply:
Ahh, okay, I can update the message.  I was worried a bit about the approach I was taking here

Reply:
I don't know if this is the best approach, unfortunately I guess there's no other option